### PR TITLE
feat: take snapshots load based

### PR DESCRIPTION
**Pull request by Zelldon**
**ZEP which described why and how:**
We should take snasphot's size based, to prevent out of disk space better and have a predictable start up time.

### CODE REVIEW

File: ZEP000-size-based-snapshot.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+We keep it like it is, but we are not able to handle high load scenarios properly. We will still
+have the issues which are pointed out above.
+
+# Prior art
+[prior-art]: #prior-art
+
+There are several raft implementation out there and I think all of them doing it in the same way, since it 
+is also described in that way in the raft paper see above. I think it was also done in atomix before.
```
Comment (deepthidevaki) [inline ZEP000-size-based-snapshot.md:138]: I think in atomix it takes snapshot periodically (or atleast attempt to take a snapshot). This period was very small - around 10sec. In addition, it also force take a snapshot if it runs out of disk space.

**Comment by Zelldon** [comment]
Hey @deepthidevaki based on our discussion we had yesterday I rewrote/improved the ZEP please have a second look. :)

**Comment by deepthidevaki** [comment]
Thanks @Zelldon I have added a summary in section "Reference level explanation". It was not clear to me why the additional flag is required. So I did not add it in the summary. Also added some questions towards the end. Other than that there are only minor changes. See the last commit. You might see big blocks of changes, but it is mostly only fixing typos or some rewording of the text. Please take a look.

**Comment by Zelldon** [comment]
@deepthidevaki it is more to avoid concurrency issues, like if the timer is triggered check whether in between there was no compaction. But yeah since we not really know whether we need that in the real implementation I can't say that. Might be that in the end we don't need it.

**Pull request merged**